src/malli/generator.cljc

```diff
@@ -48,7 +48,11 @@
 #?(:clj
    (defn -re-gen [schema opts]
      (let [re (or (first (m/childs schema opts)) (m/form schema opts))]
-       (gen2/string-from-regex (re-pattern (str/replace (str re) #"^\^?(.*?)(\$?)$" "$1"))))))
```

Comment:
I think it would be better to return `nil` in case of failure. With this, the `m/generator` would fail fast on creation (as no generator is found), not at runtime when invoked. Not perfect as the error message is the generic one, instead of the actual `test.chuck` one, but still, better IMO.

Reply:
or, actually - the `gen/elements` should be checked first, if it exists, the generator would not be looked from the schema.

Reply:
@ikitommi I've added an explicit check for `:gen/elements`. If present, nil will be returned for the generator.

Reply:
Thanks! Currently, the `:gen/elements` is already read in `-create`, could all the changes be applied there? e.g. re-order so that if the elements exists, the `-generator` would not be called at all. This would enforce the ordering of `elements` first for all schemas.

Reply:
Sure. Done.